src/server/controllers/user.js

```diff
@@ -1,35 +1,46 @@
 const bcrypt = require('bcrypt');
 const jwt = require('jsonwebtoken');
-const { PrismaClient } = require('@prisma/client');
+const { PrismaClient, Prisma } = require('@prisma/client');
 const prisma = new PrismaClient();
 
-const jwtSecret = 'mysecret';
+const jwtSecret = 'secret';
 
 const register = async (req, res) => {
     const { username, password } = req.body;
-
-    const createdUser = null;
-
+    const hash = await bcrypt.hash(password, 10)
+    try {
+        const createdUser = await prisma.user.create({
+        data: {
+            username: username,
+            password: hash
+        }
+        })
+    console.log("done")
     res.json({ data: createdUser });
+    } catch (e) {
+        if (e instanceof Prisma.PrismaClientKnownRequestError){
```

Comment:
Interesting, so this evaluates true of the error is an instance of the prisma object?

Reply:
As I understand it, Prisma has it's own type of error object - PrismaClientKnownRequestError: https://www.prisma.io/docs/concepts/components/prisma-client/handling-exceptions-and-errors

So this code catches any errors that Prisma throws up - this stops e.g. the server crashing when the client tries to add a movie that's already in the database.